layout/brandeskopf.go

```diff
@@ -158,12 +164,43 @@ func horizontalCompaction(g LayeredGraph, root map[uint64]uint64, align map[uint
 		}
 	}
 
+	// class offsets
```

Comment:
probably this should be a function. this is very complex code-block and it has comment that identifies it. -> strong signal for making this function

```suggestion
func classOffsets(...
```